pkg/runtime/reconciler.go

```diff
@@ -718,6 +717,10 @@ func (r *resourceReconciler) HandleReconcileError(
 	err error,
 ) (ctrlrt.Result, error) {
 	if ackcompare.IsNotNil(latest) {
+		// ensure that required conditions are set in latest resource before
+		// patching resource status
+		r.ensureConditions(ctx, latest, err)
```

Comment:
My only worry about this is that `handleRequeues` runs *before* `handleReconcileError` and if we aren't ensuring a ResourceSynced condition is present (with either a False or Unknown condition value) on the `latest` resource by the time that `handleRequeues` runs, then the request won't be requeued...